app/models/user.rb

```diff
@@ -0,0 +1,20 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = Settings.validates.regex.email.freeze
+
+  validates :name, presence: true,
+            length: {maximum: Settings.validates.length.max_name}
+  validates :email, presence: true,
+            length: {maximum: Settings.validates.length.max_email},
+            format: {with: VALID_EMAIL_REGEX}, uniqueness: true
+  validates :password_digest, presence: true,
+            length: {minimum: Settings.validates.length.min_pass}, if: :password
+  has_secure_password
+
+  before_save :downcase_email
+
+  private
+
+  def downcase_email
+    email.downcase!
```

Comment:
phân biệt downcase! vs downcase

Reply:
downcase! sẽ trả về nil nếu như xâu ký tự không có thay đổi.
downcase: in thường chuỗi cần